spacy/language.py

```diff
@@ -2033,8 +2034,18 @@ def replace_listeners(
             # Go over the listener layers and replace them
             for listener in pipe_listeners:
                 new_model = tok2vec_model.copy()
-                if "replace_listener" in tok2vec_model.attrs:
-                    new_model = tok2vec_model.attrs["replace_listener"](new_model)
+                replace_listener_func = tok2vec_model.attrs.get("replace_listener")
+                if replace_listener_func is not None:
+                    # Pass the extra args to the callback without breaking compatibility with
+                    # old library versions that only expect a single parameter.
+                    num_params = len(
+                        inspect.signature(replace_listener_func).parameters
+                    )
+                    assert num_params in (1, 3)
```

Comment:
I think it's nicer to have a proper exception hare. This is an attribute that downstream implementations can define. Or otherwise at least an assert with a message.

Reply:
Good point! Will fix.